public/lib/uploadHttpConcurrentProgress.js

```diff
@@ -116,8 +116,11 @@
             return;
         }
         this._collectMovingAverages = false;
-        //pushing results to an array
-        this._results.push(result);
+
+        //if request complete and no progress events then lower movingAverage
+        if(this._progressCount < this.movingAverage){
+          this.movingAverage =1;
```

Comment:
Maybe add a comment here that this is so that subsequent runs, if any, will capture more progress events. It took me a while to understand what effect changing `movingAverage` here to `1` would have.